devito/finite_differences/finite_difference.py

```diff
@@ -270,13 +272,10 @@ def indices_weights_to_fd(expr, dim, inds, weights, matvec=1):
     # Loop through weights
     terms = []
     for i, c in zip(inds, weights):
-        try:
-            iloc = i.xreplace(mapper)
-        except AttributeError:
-            iloc = i
-        subs = {expr.indices_ref[dim]: iloc}
-        terms.append(expr.subs(subs) * c)
+        iloc = sympify(i).xreplace(mapper)
```

Comment:
this block perhaps need a comment about what's going on

is the `sympify` there for the usual reasons (need for integer indices?)

Reply:
isn't try-catch (on top of your changes) potentially faster?

Reply:
>isn't try-catch (on top of your changes) potentially faster?

Not from what I tried no. WIll add comment

Reply:
That's fine, but for what is worth, I'd use a try-catch. That's what is for, and not using it in this case only makes things potentially slower.

Reply:
Actually it makes thig potentially better since it guaranties indices are already sympy object  for subsequent manipulation.

Reply:
yeah, but then you can use `sympify` only in the catch? so eventually you systematically have sympy objects